cmd/installer/cli/upgrade.go

```diff
@@ -0,0 +1,439 @@
+
+	return nil
+}
+
+func addUpgradeManagementConsoleFlags(cmd *cobra.Command, flags *UpgradeCmdFlags) error {
+	// default value of 0 indicates no user input - will use existing runtime config value
+	cmd.Flags().IntVar(&flags.managerPort, "manager-port", 0, "Port on which the Manager will be served")
+	return nil
+}
+
+func preRunUpgrade(ctx context.Context, flags UpgradeCmdFlags, upgradeConfig *upgradeConfig, rc runtimeconfig.RuntimeConfig, kcli client.Client) error {
+	// TODO: Add check for kubernetes target when kubernetes support is added
+	if !slices.Contains([]string{"linux"}, flags.target) {
+		return fmt.Errorf(`invalid --target (must be: "linux")`)
+	}
+
+	// Verify an installation exists and get the cluster ID
+	clusterID, err := getClusterID(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("could not get existing installation: %w", err)
+	}
+	upgradeConfig.clusterID = clusterID
+
+	// Verify that a data directory exists
+	dataDir := rc.EmbeddedClusterHomeDirectory()
+	if _, err := os.Stat(dataDir); os.IsNotExist(err) {
+		return fmt.Errorf("upgrade requires existing data directory from previous installation")
+	}
+
+	// Validate the license is indeed a license file
+	license, err := getLicenseFromFilepath(flags.licenseFile)
+	if err != nil {
+		return err
+	}
+	upgradeConfig.license = license
+	data, err := os.ReadFile(flags.licenseFile)
+	if err != nil {
+		return fmt.Errorf("unable to read license file: %w", err)
+	}
+	upgradeConfig.licenseBytes = data
+
+	if flags.airgapBundle != "" {
+		metadata, err := airgap.AirgapMetadataFromPath(flags.airgapBundle)
+		if err != nil {
+			return fmt.Errorf("failed to get airgap info: %w", err)
+		}
+		upgradeConfig.airgapMetadata = metadata
+	}
+
+	assetsSize, err := goods.SizeOfEmbeddedAssets()
+	if err != nil {
+		return fmt.Errorf("failed to get size of embedded files: %w", err)
+	}
+	upgradeConfig.embeddedAssetsSize = assetsSize
+
+	// Read existing TLS certificates from kotsadm-tls secret in the cluster
+	tlsConfig, err := readTLSConfig(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("upgrade requires TLS certificates from existing installation: %w", err)
+	}
+	upgradeConfig.tlsConfig = tlsConfig
+	cert, err := tls.X509KeyPair(upgradeConfig.tlsConfig.CertBytes, upgradeConfig.tlsConfig.KeyBytes)
+	if err != nil {
+		return fmt.Errorf("failed to create TLS certificate from data: %w", err)
+	}
+	upgradeConfig.tlsCert = cert
+
+	// Read password hash from the kotsadm-password secret in the cluster
+	pwdHash, err := readPasswordHash(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("failed to read password hash from cluster: %w", err)
+	}
+	upgradeConfig.passwordHash = pwdHash
+
+	if os.Getuid() != 0 {
+		return fmt.Errorf("upgrade command must be run as root")
+	}
+
+	// set the umask to 022 so that we can create files/directories with 755 permissions
+	// this does not return an error - it returns the previous umask
+	_ = syscall.Umask(0o022)
```

Comment:
I think we should move this to the top of the function in case in the future anything gets added above it that creates a directory or file